### PR TITLE
added useContext, badge count icon

### PR DESCRIPTION
**Pull request by Sosnovskyi-Roman**

### CODE REVIEW

File: src/components/Header/Search/search.module.scss [hunk 1]
```diff
@@ -24,9 +24,41 @@
     }
 
     @media (width <= 470px) {
-      width: auto;
+      height: 48px;
```
Comment (bessu4ka) [inline src/components/Header/Search/search.module.scss:27]: This property has no effect, please remove it.

File: src/components/Header/Search/search.module.scss [hunk 2]
```diff
@@ -24,9 +24,41 @@
...
+    flex-direction: row;
+    justify-content: center;
+    align-items: center;
+  }
+
+  span {
+    width: 48px;
+    height: 48px;
+    border-radius: 0px 12px 12px 0px;
+    background: rgb(0, 130, 13);
```
Comment (bessu4ka) [inline src/components/Header/Search/search.module.scss:60]: `background : var(--green-color);`

File: src/components/Header/header.module.scss [hunk 3]
```diff
@@ -76,4 +85,28 @@
     padding-bottom: 16px;
     background-color: var(--white-color);
   }
+  .badgeCount {
+    position: absolute;
+    color: white;
```
Comment (bessu4ka) [inline src/components/Header/header.module.scss:90]: `color: var(--white-color);`

File: src/components/Header/header.module.scss [hunk 4]
```diff
@@ -76,4 +85,28 @@
...
+    width: 18px;
+    height: 18px;
+    border-radius: 39px;
```
Comment (bessu4ka) [inline src/components/Header/header.module.scss:93]: Why specifically 39?...
image
simplify code

```
.badgeCount {
    top: 30px;
    left: 24px;
    color: var(--white-color);
    padding: 4px;
    position: absolute;
    border-radius: 10px;
    background: var(--theme-color);
  }
```
image
 its just example, U can do code better :)

File: src/components/Header/header.module.scss [hunk 5]
```diff
@@ -76,4 +85,28 @@
...
+    align-items: center;
+    background: rgb(111, 76, 154);
+    bottom: -6px;
+    right: -2px;
+  }
+}
+
+@media (max-width: 390px) {
+  .headerBottom {
+    padding-right: 16px;
```
Comment (bessu4ka) [inline src/components/Header/header.module.scss:107]: Simplify code

File: src/pages/Layout/index.tsx [hunk 6]
```diff
@@ -2,11 +2,32 @@ import { Outlet } from 'react-router-dom';
 import Header from '@/components/Header/Header';
 import Footer from '@/components/footer/footer';
 import style from './Layout.module.scss';
+import { useState } from 'react';
+import BurgerMenu from '@/components/BurgerMenu/BurgerMenu.tsx';
+import { MenuContext } from '@/context/menuContext.ts';
 
 const Layout = () => {
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
```
Comment (bessu4ka) [inline src/pages/Layout/index.tsx:10]: This is extra typing; TypeScript types primitive values on its own. You can verify this by removing the code.
`const [isMenuOpen, setIsMenuOpen] = useState(false);`

File: src/components/BurgerMenu/BurgerMenu.tsx [hunk 7]
```diff
@@ -1,35 +1,38 @@
...
     <>
-      <Button onClick={showDrawer} type="text">
-        <img src={BurgerMenuIcon} />
-      </Button>
-
       <Drawer
         closable={false}
-        onClose={closeDrawer}
-        open={open}
+        width={'100vh'}
```
Comment (bessu4ka) [inline src/components/BurgerMenu/BurgerMenu.tsx:19]: `vh` its height (vertical), mayby you mean `vw`?
Please check the code, as I don't fully understand what it's supposed to do at first glance.

File: src/components/BurgerMenu/BurgerMenu.tsx [hunk 8]
```diff
@@ -1,35 +1,38 @@
...
 import items from './constants';
 import ButtonNav from '../Button/Button';
 import buttonData from '@/constants/ButtonConstants';
+import { Header } from '../components';
+import { useMenuContext } from '@/context/menuContext.ts';
 
 export default function BurgerMenu() {
-  const [open, setOpen] = useState(false);
+  //const [open, setOpen] = useState(false);
+  const { onMenuClose, isMenuOpen } = useMenuContext();
```
[the remark this answers is not in the document]
Comment (Sosnovskyi-Roman) [reply src/components/BurgerMenu/BurgerMenu.tsx:13]: I needed to use the state manager,useState is only available in the component where it is declared. If we need to share this state, we'll have to pass it as props to the child components, which will result in drilling props,useContext allows state or data to be available to any child component within Context.Provider without having to pass properties explicitly.

File: src/components/BurgerMenu/BurgerMenu.tsx [hunk 9]
```diff
@@ -1,35 +1,38 @@
...
         closable={false}
-        onClose={closeDrawer}
-        open={open}
+        width={'100vh'}
+        onClose={onMenuClose}
+        open={isMenuOpen}
         placement="left"
+        bodyStyle={{ padding: '0', overflowY: 'auto' }}
       >
         <div>
```
[the remark this answers is not in the document]
Comment (Sosnovskyi-Roman) [reply src/components/BurgerMenu/BurgerMenu.tsx:25]: Yes

**Comment by bessu4ka** [comment]
Please, pull main branch and resolve the conflicts
image

**Review by Rozygeldi94 — APPROVED** [review]

**Pull request merged**